px4_offboard/visualizer.py

```diff
@@ -118,10 +118,20 @@ def __init__(self):
         # trail size
         self.trail_size = 1000
 
+        # time stamp for the last local position update
+        self.last_local_pos_update = 0.0
+        # time after which existing path is cleared upon receiving new
+        # local position ROS2 message
+        self.path_clearing_timeout_ms = 1000
```

Comment:
Would there be a way to disable this by default? I just don't want some situation that the poses are getting removed without the user being aware of it.

Reply:
Sure, I see couple ways to introduce on/off switching with different levels of user involvement and interactivity during simulation run. 

- Option number 1:

One way to set this feature on/off once per visualizer node if to fix timeout value to enormously big number, such as maximum value that `float` type can hold, and leave information in `README.md` for user to change it to considerably lower value, such as 1000 which corresponds to path clearing if time difference between last local position update is at least 1000 ms regarding previous one. Maximum value of float returned by
```python
import sys
print(sys.float_info.max)
```
on my system  code above yields `1.7976931348623157e+308` which is more than enough milliseconds not to accidently clear path between consequent SITL runs.

- Option number 2:

Another way is to declare ROS2 parameter using line below (defaulting to `False`) which allows interactions while visualizer is running

```python
self.declare_parameter("clear_path_on_sitl_restart", False)
```
one then could in runtime call
```sh
ros2 param set /px4_offboard/visualizer clear_path_on_sitl_restart True
```
and from then on, old path would be cleared upon SITL restarting. This one is perhaps, a little inconvenient but does not require user to modify code at all, with downside of having to manually call `ros2 param set` every time visualizer is started, if he is interested in that feature. This option is sweetspot between amount of code required to implement the feature and user involvement in setting it.

- Option number 3:

The last option I thought about is to utilize ROS2 launch parameters propagated into system automatically through launch file, which would allow user to modify default value in yaml parameter file, but that would include extending repository by additional file (which is not a problem on its own but for new users the amount of files could be an indicator whether repository is comprehendible easily or needs more analysis).

Please let me know what do you think about my ideas, also please note that even without this feature old path is still slowly being erased even if SITL simulation is killed leaving user with finite amount of time for analysis before it disappears, as new path points are computed and sent in timer callback, regardless whether local position updates are coming or not. The feature also clears path only upon next SITL run, not on current SITL kill.

Reply:
I think I would simply set  self.path_clearing_timeout_ms as a `nan`, and only timeout when you define it through a parameter. Does this make sense?